replication/row_event.go

```diff
@@ -985,6 +994,29 @@ func (e *RowsEvent) DecodeHeader(data []byte) (int, error) {
 	return pos, nil
 }
 
+func (e *RowsEvent) decodeExtraData(data []byte) (err2 error) {
+	pos := 0
+	extraDataType := data[pos]
+	pos += 1
+	switch extraDataType {
+	case ENUM_EXTRA_ROW_INFO_TYPECODE_NDB:
+		var ndbLength int = int(data[pos])
+		pos += 1
+		e.NdbFormat = data[pos]
+		pos += 1
+		e.NdbData = data[pos : pos+ndbLength-2]
+	case ENUM_EXTRA_ROW_INFO_TYPECODE_PARTITION:
+		if e.eventType == UPDATE_ROWS_EVENTv2 || e.eventType == PARTIAL_UPDATE_ROWS_EVENT {
```

Comment:
I'm still not sure about it. MySQL enables it for `UPDATE_ROWS_EVENT`, `UPDATE_ROWS_EVENT_V1` and `PARTIAL_UPDATE_ROWS_EVENT`, and we do it for `UPDATE_ROWS_EVENTv2` and `PARTIAL_UPDATE_ROWS_EVENT`. I remember `UPDATE_ROWS_EVENT` and `UPDATE_ROWS_EVENT_V1` has been deprecated so we really don't need to care about them. I'll check it later.

And it's helpful that you can also help me check and provide some reference 😄

Reply:
As you says in PR description, this feature is added in v8.0.16, and [log_bin_use_v1_row_events](https://dev.mysql.com/doc/refman/8.0/en/replication-options-binary-log.html#sysvar_log_bin_use_v1_row_events) is deprecated in v8.0.18, so we should also consider UPDATE_ROWS_EVENT_V1.

And MySQL's UPDATE_ROWS_EVENT (`31`) is our UPDATE_ROWS_EVENTv2